packages/data-service/src/data-service.ts

```diff
@@ -2285,6 +2315,74 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return { name, ...normalized };
   }
 
+  @op(mongoLogId(1_001_000_063))
+  async previewUpdate(
+    ns: string,
+    filter: Document,
+    update: Document,
+    executionOptions?: ExecutionOptions
+  ): Promise<UpdatePreview> {
+    if (!executionOptions) {
+      const controller = new AbortController();
+      executionOptions = { abortSignal: controller.signal };
```

Comment:
executionOptions.abortSignal is optional, so unless you need an abortSignal below you don't have to do this.